src/run.rs

```diff
@@ -1,38 +1,47 @@
 use chrono::prelude::*;
 use clap::ArgMatches;
 
-use crate::util::{
-    file::{cargo_path, day_path, download_input_file},
-    get_year,
-    submit::{self, get_submit_day},
+use crate::{
+    error::AocError,
+    util::{
+        file::{cargo_path, day_path, download_input_file},
+        get_year,
+        submit::{self, get_submit_day},
+    },
 };
 
-pub fn get_day(matches: &ArgMatches) -> Result<u32, std::num::ParseIntError>
+pub fn get_day(matches: &ArgMatches) -> Result<u32, AocError>
 {
-    matches.get_one::<String>("day").unwrap().parse()
+    let day = matches
+        .get_one::<String>("day")
+        .ok_or(AocError::ArgMatches)?
+        .parse::<u32>()
+        .map_err(|_| AocError::ParseNum)?;
```

Comment:
Its picky, but can `Into` be implemented here instead?

Reply:
I agree, it's used enough places to make it worth it